main.tf

```diff
@@ -1,14 +1,15 @@
 data "aws_caller_identity" "default" {}
 
-# Make a topic
 resource "aws_sns_topic" "default" {
-  name_prefix = "${local.alert_for}-threshold-alerts"
+  count = module.this.enabled ? 1 : 0
+  name  = join(module.this.delimiter, [local.alert_for, "threshold", "alerts"])
+  tags  = module.this.tags
 }
 
 resource "aws_sns_topic_policy" "default" {
-  count  = "${var.add_sns_policy != "true" && var.sns_topic_arn != "" ? 0 : 1}"
-  arn    = "${local.sns_topic_arn}"
-  policy = "${data.aws_iam_policy_document.sns_topic_policy.json}"
+  count  = module.this.enabled != true && var.add_sns_policy != "true" && var.sns_topic_arn != "" ? 0 : 1
```

Comment:
Huh, I've never seen a count / for_each expression that is enabled on the negative like this... but it was like this before you did your updates so let's let it ride.